File: 2187-minimum-time-to-complete-trips/2187-minimum-time-to-complete-trips.cpp
class Solution {
public:
    long long minimumTime(vector<int>& time, int totalTrips) {
        long long ans = 0;
        sort(time.begin(),time.end());
        unsigned long long low = 1,high = 1LL * totalTrips*time[0];
        while(low<high){
            long mid = (low+high)>>1;
            long total = 0;
            for(auto i:time){
                total += mid/i;
                if(total>=totalTrips)
                    break;
            }
            if(total < totalTrips)
                low = mid+1;
            else
                high = mid;
        }
        
        return high;
    }
};

File: 2187-minimum-time-to-complete-trips/2187-minimum-time-to-complete-trips.py
class Solution:
    def minimumTime(self, time: List[int], totalTrips: int) -> int:
        ans = 0;
        time.sort()
        low ,high = 1,totalTrips*time[0];
        while(low<high):
            mid = low + (high-low)//2;
            total = 0
            for i in time:
                total += mid//i
            if total < totalTrips:
                low = mid+1
            else:
                high = mid
        return high;